gammapy/maps/wcsnd.py

```diff
@@ -355,17 +355,77 @@ def _reproject_hpx(self, geom, mode='interp', order=1):
 
         return map_out
 
-    def pad(self, pad_width):
-        raise NotImplementedError
+    def pad(self, pad_width, mode='edge', cval=0):
```

Comment:
This pad method is pretty long and hard to read. I only had a very quick look and it isn't obvious to me where the default case mode='edge' is handled. maybe it would help to split this out into two methods that represent the if / else? I'm not sure it's useful to have the single `pad` that then dispatches to the two different algorithms, but if so, then the other two methods could be private (leading `_`). Also needs a docstring.

Reply:
Ok I can try to split this out into separate private methods.  Regarding the docstring all of the abstract methods inherit their docstring from the `Map` class so that's why no docstring appears here.